src/applications/appeals/shared/components/ContestableIssues.jsx

```diff
@@ -70,17 +79,40 @@ const ContestableIssues = props => {
   // inReviewMode = true (review page view, not in edit mode)
   // inReviewMode = false (in edit mode)
   const inReviewMode = (onReviewPage && formContext.reviewMode) || false;
-  const showCheckbox = !onReviewPage || (onReviewPage && !inReviewMode);
   const { submitted } = formContext;
-  const loadedIssues = formData.contestedIssues || [];
+  const loadedIssues = useMemo(() => formData.contestedIssues || [], [
+    formData.contestedIssues,
+  ]);
+
+  // Evaluate issues for same-day decision date blocking (computed at render time)
+  const issuesWithDateEvaluation = useMemo(
+    () =>
+      loadedIssues.map(issue => {
+        const approxDecisionDate = issue?.attributes?.approxDecisionDate;
+        const decisionDate = parseDateToDateObj(
+          approxDecisionDate,
+          FORMAT_YMD_DATE_FNS,
+        );
+        let blockingType = null;
 
-  // combine all issues for viewing
-  const items = loadedIssues
-    .map(item => ({
-      ...item?.attributes,
-      [SELECTED]: item?.[SELECTED],
-    }))
-    .concat(formData.additionalIssues || []);
+        const isBlockedSameDay = isTodayOrInFuture(decisionDate);
+        if (isBlockedSameDay) {
+          blockingType = isLocalToday(decisionDate) ? 'local' : 'utc';
```

Comment:
It's fine to leave this as is, but as a nitpick, I think it would be more efficient to refactor `isTodayOrInFuture` to return the `blockingType` directly since the main return for `isTodayOrInFuture` is just :
`return isLocalToday(date) || isUTCTodayOrFuture(date);` 
And then you check isLocalToday again in the next line to determine the blockingType. 

Alternatively, we could just use the isLocalToday and isUTCTodayOrFuture methods separately.